src/lib/util/timeconv.h

```diff
@@ -14,18 +14,232 @@
+			tm_conversion_clock::base_second,
+			tm_conversion_clock::period>(tp);
+		return internal_to_tm(normalized_tp.time_since_epoch());
+	}
+
+
+	//---------------------------------------------------------
+	//	to_system_clock - converts to a system_clock time_point
+	//---------------------------------------------------------
+
+	static std::chrono::time_point<std::chrono::system_clock> to_system_clock(const time_point &tp)
+	{
+		auto normalized_tp = to_arbitrary_time_point<
+			std::int64_t,
+			system_conversion_clock::base_year,
+			system_conversion_clock::base_month,
+			system_conversion_clock::base_day,
+			system_conversion_clock::base_hour,
+			system_conversion_clock::base_minute,
+			system_conversion_clock::base_second,
+			system_conversion_clock::period>(tp);
+		return std::chrono::time_point<std::chrono::system_clock>(normalized_tp.time_since_epoch() + system_clock_adjustment);
+	}
+
+	//---------------------------------------------------------
+	//	from_system_clock - converts from a system_clock time_point
+	//---------------------------------------------------------
+
+	static time_point from_system_clock(const std::chrono::time_point<std::chrono::system_clock> &tp)
+	{
+		std::chrono::time_point<system_conversion_clock> normalized_tp(tp.time_since_epoch() - system_clock_adjustment);
+		return from_arbitrary_time_point(normalized_tp);
+	}
+
+private:
+	// By positioning the base year at 1601, we can ensure that:
+	//
+	//   * years with leap years are at the end of every quadyear
+	//   * quadyears without leap years are at the end of every century
+	//	 * centuries where the last quadyear has a leap year at the end are at the
+	//	   end of every quadcentury
+	typedef arbitrary_clock<std::int64_t, 1601, 1, 1, 0, 0, 0, std::ratio<1, 1> > tm_conversion_clock;
+
+	//---------------------------------------------------------
+	//	internal_to_tm - formats a structure of type 'struct tm'
+	//	based on a normalized clock
+	//---------------------------------------------------------
+
+	static struct tm internal_to_tm(std::chrono::duration<std::int64_t, std::ratio<1, 1> > duration)
+	{
+		constexpr auto days_in_year(365);
+		constexpr auto days_in_four_years((days_in_year * 4) + 1);
+		constexpr auto days_in_century((days_in_four_years * 25) - 1);
+		constexpr auto days_in_four_centuries((days_in_century * 4) + 1);
+
+		constexpr tm_conversion_clock::duration day(std::chrono::hours(24));
+		constexpr tm_conversion_clock::duration year(day * days_in_year);
+		constexpr tm_conversion_clock::duration four_years(day * days_in_four_years);
+		constexpr tm_conversion_clock::duration century(day * days_in_century);
+		constexpr tm_conversion_clock::duration four_centuries(day * days_in_four_centuries);
+
+		// figure out the day of week (note that 0 is Sunday, but January 1st 1601 is
+		// a Monday, so we have to adjust by one day)
+		int day_of_week = (int)(((duration + std::chrono::hours(24)) / day) % 7);
```

Comment:
many things here can be `const` and many parentheses are unnecessary because function-style cast syntax